packages/trpc/server/routers/viewer/availability/schedule/get.handler.ts

```diff
@@ -94,18 +94,27 @@ export const getHandler = async ({ ctx, input }: GetOptions) => {
           .minute(override.endTime.getUTCMinutes())
           .toDate(),
       };
-      const dayRangeIndex = acc.findIndex(
-        // early return prevents override.date from ever being empty.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        (item) => yyyymmdd(item.ranges[0].start) === yyyymmdd(override.date!)
+
+      // Create a new date string for the current override
+      // early return prevents override.date from ever being empty.
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const overrideDateString = yyyymmdd(override.date!);
+
+      // Check if there's a range with the day of the override date, to append a new range to it
+      const existingDayIndex = acc.findIndex(
+        (item) => yyyymmdd(dayjs(item.day)) === dayjs(override.date).format("YYYY-MM-DD")
```

Comment:
This was causing a bug that would put the range time of a consecutive day in a previous one. For example, 18th 9:00AM - 5:00PM and 19th 9:00AM - 5:00PM would be after this 18th 9:00AM -5:00PM (x2), and 19th would have gone away.